CMakeLists.txt

```diff
@@ -0,0 +1,256 @@
+option(BUILD_TEST "builds the tests" ON)
+option(USE_EXTERNAL_MAPBOX_GEOMETRY "Use a external mapnik/geometry.hpp. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_POLYLABEL "Use a external mapnik/polylabel. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_PROTOZERO "Use a external mapnik/protozero. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_VARIANT "Use a external mapnik/variant. If off, use the submodule" OFF)
+option(USE_JPEG "adds jpeg support" ON)
+option(USE_PNG "adds png support" ON)
+option(USE_TIFF "adds tiff support" ON)
+option(USE_WEBP "adds webp support" ON)
+option(USE_LIBXML2 "adds libxml2 support" ON)
+option(USE_CAIRO "adds the cairo renderer" ON)
+option(USE_PROJ4 "adds proj4 support" ON)
+option(USE_GRID_RENDERER "adds grid renderer" ON)
+option(USE_SVG_RENDERER "adds grid renderer" ON)
+
+option(USE_PLUGIN_INPUT_CSV "adds plugin input csv" ON)
+option(USE_PLUGIN_INPUT_GDAL "adds plugin input gdal" ON)
+option(USE_PLUGIN_INPUT_GEOBUF "adds plugin input geobuf" ON)
+option(USE_PLUGIN_INPUT_GEOJSON "adds plugin input geojson" ON)
+option(USE_PLUGIN_INPUT_OGR "adds plugin input ogr" ON)
+option(USE_PLUGIN_INPUT_PGRASTER "adds plugin input pgraster" ON)
+option(USE_PLUGIN_INPUT_POSTGIS "adds plugin input postgis" ON)
+option(USE_PLUGIN_INPUT_RASTER "adds plugin input raster" ON)
+option(USE_PLUGIN_INPUT_SHAPE "adds plugin input shape" ON)
+option(USE_PLUGIN_INPUT_SQLITE "adds plugin input sqlite" ON)
+option(USE_PLUGIN_INPUT_TOPOJSON "adds plugin input topojson" ON)
+
+option(BUILD_DEMO_VIEWER "builds the demo viewer" ON)
+option(BUILD_DEMO_CPP "builds the demo c++ application" ON)
+
+option(BUILD_UTILITY_GEOMETRY_TO_WKB "builds the utility program geometry_to_wkb" ON)
+option(BUILD_UTILITY_MAPNIK_INDEX "builds the utility program mapnik_index" ON)
+option(BUILD_UTILITY_MAPNIK_RENDER "builds the utility program mapnik_render" ON)
+option(BUILD_UTILITY_OGRINDEX "builds the utility program ogrindex" OFF)
+option(BUILD_UTILITY_PGSQL2SQLITE "builds the utility program pgsql2sqlite" ON)
+option(BUILD_UTILITY_SHAPEINDEX "builds the utility program shapeindex" ON)
+option(BUILD_UTILITY_SVG2PNG "builds the utility program svg2png" ON)
+
+set(PLUGINS_INSTALL_DIR bin/plugins/input CACHE STRING "installs the plugins in the specified directory")
+get_filename_component(ABSOLUTE_PLUGINS_INSTALL_DIR ${PLUGINS_INSTALL_DIR} ABSOLUTE)
+message("Installing plugins to ${ABSOLUTE_PLUGINS_INSTALL_DIR}")
+set(FONTS_INSTALL_DIR bin/fonts CACHE STRING "installs the fonts in the specified directory")
+get_filename_component(ABSOLUTE_FONTS_INSTALL_DIR ${FONTS_INSTALL_DIR} ABSOLUTE)
+message("Installing fonts to ${ABSOLUTE_FONTS_INSTALL_DIR}")
+
+set(CMAKE_CXX_STANDARD 14 CACHE STRING "Sets the c++ standard. c++11 is minimum.")
+message("Using c++${CMAKE_CXX_STANDARD}")
+
+find_package(Boost 1.74 REQUIRED COMPONENTS filesystem system regex)
+find_package(ICU REQUIRED COMPONENTS uc)
```

Comment:
```diff
-find_package(ICU REQUIRED COMPONENTS uc)
+find_package(ICU REQUIRED COMPONENTS uc data i18n)
.....
+set_property(TARGET Boost::regex PROPERTY INTERFACE_LINK_LIBRARIES "") # remove transitive dependencies
+
 target_link_libraries(headers INTERFACE
-    Boost::boost
+  Boost::headers
   Boost::regex
   Boost::filesystem
   ICU::uc
+  ICU::data
+  ICU::i18n
   Freetype::Freetype
   harfbuzz::harfbuzz
   ${MAPNIK_OPTIONAL_LIBS}
 )
+
```

Reply:
i'm aware `Boost::boost` is alias to `Boost::headers`, I just prefer the naming to reflect the intent.

Reply:
On the same note, perhaps we could come up with a better name for 'mapnik::headers' target ?

Reply:
agree with that. will change that 


Maybe mapnik::core? since these headers are needed in all projects. Mainly because of the config, version and warning headers.

but it might also be worth a thought to move them to a separate project and leave the rest of the headers, which are only needed for mapnik::mapnik, in include.

Reply:
For the PROJ naming: Maybe we should change the define `MAPNIK_USE_PROJ4 `to `MAPNIK_USE_PROJ `?

Reply:
changed with 1f73d1c